src/Domain/Creators/DomainCreator.hpp

```diff
@@ -10,6 +10,7 @@
 #include <string>
 
 #include "DataStructures/Tensor/IndexType.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
```

Comment:
Is this include really needed?  Also not your fault, but the includes and forward declarations for this file are completely messed up.  Can you fix them while touching this file?  If it turns out to be a mess because of missing includes in files that included this one, then maybe just file an issue

Reply:
I believe it is needed to destroy the temporary returned from `functions_of_time()`.  (That function is where I got errors, in any case.)

I'll take a look at the other includes.

Reply:
ah yes, that makes sense